services/ui-src/src/components/reports/StandardReportPage.tsx

```diff
@@ -58,6 +59,42 @@ export const StandardReportPage = ({ route, validateOnRender }: Props) => {
     navigate(nextRoute);
   };
 
+  const filterResubmissionData = (formData: FormJson) => {
+    //make deep copies of formData
+    const formDataCopy = JSON.parse(JSON.stringify(formData));
+    const formDataFields = JSON.parse(JSON.stringify(formData?.fields));
+
+    if (
+      report?.reportType === "SAR" &&
+      report?.submissionCount! === 1 &&
+      report?.status === ReportStatus.SUBMITTED
+    ) {
+      const removeResubmissionDataFields: any[] = formDataFields.filter(
+        (data: AnyObject) =>
+          data.id !== "generalInformation_resubmissionInformation"
+      );
+      formDataCopy.fields = [...removeResubmissionDataFields];
+
+      return formDataCopy;
+    } else if (
+      report?.reportType === "SAR" &&
+      report?.submissionCount! >= 1 &&
+      (report?.status === ReportStatus.IN_REVISION ||
+        report?.status === ReportStatus.SUBMITTED ||
+        report?.status === ReportStatus.IN_PROGRESS)
+    ) {
+      return formDataCopy;
+    } else {
+      const removeResubmissionDataFields: any[] = formDataFields.filter(
+        (data: AnyObject) =>
+          data.id !== "generalInformation_resubmissionInformation"
+      );
+      formDataCopy.fields = [...removeResubmissionDataFields];
+
+      return formDataCopy;
```

Comment:
the else and the first if now have the same logic, right?

Reply:
Same logic but different condition. I could put it in a function..